### DISCUSSION TITLE
How to enable zlibopenssh.com compression?

### DISCUSSION THREAD

**Discussion by geniy21** [sshnet/SSH.NET#1431] [Q&A] [answered]
Hello! 

A new release [2024.1.0](https://github.com/sshnet/SSH.NET/releases/tag/2024.1.0) has a new feature zlib@openssh.com compression, which was added in this [PR1326](https://github.com/sshnet/SSH.NET/pull/1326). Now I'd like to use  zlib@openssh.com compression algorithm in my code, but how to do that? I used the unit tests as a reference, but the same idea will not work as ZlibOpenSsh is marked as internal class and cannot be accessible anymore:
`conn.CompressionAlgorithms.Clear();
 conn.CompressionAlgorithms.Add(new KeyValuePair<string, Func<Compressor>>("zlib@openssh.com", () => new ZlibOpenSsh()));`
Another idea is to remove "none" from the dictionary: 
`conn.CompressionAlgorithms.Remove("none")`
then there will be only "zlib@openssh.com" algorithm left and it will be used as default I assume. Even if this approach might work, I don't like it as I want explicitly set compression algorithm somehow. Any help is appreciated. Thanks in advance!

**Comment by Rob-Hague** [comment, marked as answer]
Yeah, that would be the way to enforce it. The idea was that it would be used if the server required it, but we should probably make it public so one can add it explicitly as you tried

**Comment by geniy21** [comment]
I think this is a way how you can make zlib@openssh.com as default:
`const string zlibKey = "zlib@openssh.com";
  if (conn.CompressionAlgorithms.TryGetValue(zlibKey, out var compressor))
  {
      conn.CompressionAlgorithms.Clear();
      conn.CompressionAlgorithms.Add(new KeyValuePair<string, Func<Compressor>>(zlibKey, compressor));
      conn.CompressionAlgorithms.Add(new KeyValuePair<string, Func<Compressor>>("none", null));
  }`
So, now zlib@openssh.com be the first and preferred compression algorithm, a bit "hacky" way, but it should work
